.generator/src/generator/templates/client.j2

```diff
@@ -115,35 +117,89 @@ func ParameterToString(obj interface{}, collectionFormat string) string {
 {# The method is used in Terraform client and needs to be public. -#}
 // CallAPI do the request.
 func (c *APIClient) CallAPI(request *http.Request) (*http.Response, error) {
-	if c.Cfg.Debug {
-		dump, err := httputil.DumpRequestOut(request, true)
-		if err != nil {
-			return nil, err
+	var rawBody []byte
+	if request.Body != nil && request.Body != http.NoBody {
+		rawBody, _ = io.ReadAll(request.Body)
+		request.Body.Close()
+	}
+
+	ctx, ccancel := context.WithTimeout(request.Context(), c.Cfg.RetryConfiguration.HTTPRetryTimeout)
+	defer ccancel()
+	maxRetries := c.Cfg.RetryConfiguration.MaxRetries
+	retryCount := 0
+	for {
+		if retryCount == maxRetries {
+			ccancel()
 		}
-		// Strip any api keys from the response being logged
-		keys, ok := request.Context().Value(ContextAPIKeys).(map[string]APIKey)
-		if keys != nil && ok {
-			for _, apiKey := range keys {
-				valueRegex := regexp.MustCompile(fmt.Sprintf("(?m)%s", apiKey.Key))
-				dump = valueRegex.ReplaceAll(dump, []byte("REDACTED"))
+		newRequest := copyRequest(request, &rawBody)
+		if c.Cfg.Debug {
+			dump, err := httputil.DumpRequestOut(newRequest, true)
+			if err != nil {
+				return nil, err
+			}
+			// Strip any api keys from the response being logged
+			keys, ok := newRequest.Context().Value(ContextAPIKeys).(map[string]APIKey)
+			if keys != nil && ok {
+				for _, apiKey := range keys {
+					valueRegex := regexp.MustCompile(fmt.Sprintf("(?m)%s", apiKey.Key))
+					dump = valueRegex.ReplaceAll(dump, []byte("REDACTED"))
+				}
 			}
+			log.Printf("\n%s\n", string(dump))
+		}
+		resp, requestErr := c.Cfg.HTTPClient.Do(newRequest)
+
+		if requestErr != nil {
+			return resp, requestErr
+		}
+
+		if c.Cfg.Debug {
+			dump, _ := httputil.DumpResponse(resp, true)
+			log.Printf("\n%s\n", string(dump))
+			fmt.Println("Max retries:", maxRetries, " Current retry:", retryCount)
+			if retryCount == maxRetries {
+				fmt.Println("Max retries reached")
```

Comment:
I think you should remove this, and log in `!shouldRetry` under.

Reply:
By turning debug on we should dump out the response no matter retry or not

Reply:
Yeah I'm just talking about the last log line,